keps/sig-storage/20190204-kubernetes-csi-release-process.md

```diff
@@ -0,0 +1,233 @@
+  deployed in a cluster
+- define a release process for each component and a "kubernetes-csi release"
+
+### Non-Goals
+
+- change responsibilities among the kubernetes-csi maintainers - as before, each
+  component will have some main maintainer who is responsible for releasing updates
+  of that component
+- automatically generate release notes
+- automatically generate "kubernetes-csi release" artifacts - this will need further thoughts
+
+
+## Proposal
+
+### Versionioning
+
+Each of the components has its own documentation, versioning and
+release notes. [Semantic versioning](https://semver.org/) is used for
+components that provide a stable API, like for example:
+- external-attacher
+- external-provisioner
+- external-snapshotter
+- cluster-driver-registrar
+- node-driver-registrar
+- csi-driver-host-path
+- csi-lib-utils
+
+Other components are internal utility packages that are
+not getting tagged:
+- csi-release-tools
+
+A "kubernetes-csi release" is a specific set of component
+releases. It's also called the "combined release". Documentation for
+combined releases is found on [kubernetes-csi
+docs](https://kubernetes-csi.github.io/docs/). The combined release is
+typically going to be prepared and announced less frequently than the
+individual component releases. It is therefore only a recommendation
+that downstream users use this combination of components. In
+particular, individual components might also get minor updates after a
+kubernetes-csi release without updating that combined release.
+
+The
+[hostpath example deployment](https://github.com/kubernetes-csi/csi-driver-host-path/tree/master/deploy)
+defines the components that are part of a combined release. This
+implies that the hostpath driver repo must be updated and tagged to
+create a new combined release. Therefore the hostpath driver's version
+becomes the kubernetes-csi release version, which is increased
+according to the same rules as the individual components (major
+version bump when any of its components had a major change, etc.).
+
+#### Release artifacts
+
+Tagging a component with a semantic version number triggers a release
+build for that component. The output is primarily the container image
+for components that need to be deployed. Those images get published as
+`gcr.io/kubernetes-csi/<image>:<tag>` pending [issue
+158](https://github.com/kubernetes/k8s.io/issues/158).
+
+Only binaries provided as part of such a release should be considered
+production ready. Binaries are never going to be rebuilt, therefore an
+image like `csi-node-driver-registrar:v1.0.2` will always pull the
+same content and `imagePullPolicy: Always` can be omitted.
+
+Eventually auxiliary files like the RBAC rules that are included in
+each component might also get published in a single location, like
+`https://dl.k8s.io/kubernetes-csi/`, but this currently outside of
+this KEP.
+
+#### Release process
+
+* A change is submitted against master.
```

Comment:
Can you make this a numbered list since there is an ordering between these steps?